src/simulators/statevector/statevector_state.hpp

```diff
@@ -634,18 +634,20 @@ void State<statevec_t>::snapshot_matrix_expval(const Operations::Op &op,
       first = false;
     else
       BaseState::qreg_.revert(true);
-
     // Apply each matrix component
     for (const auto &pair: param.second) {
-      const reg_t &qubits = pair.first;
+      reg_t sub_qubits;
+      for (const auto pos : pair.first) {
+        sub_qubits.push_back(qubits[pos]);
```

Comment:
`pos` is not linear, right? ... otherwise a `std::copy()` is preferable for copying ranges of vectors

Reply:
`pos` could be any integers. Eg `[0, 3, 7]`